app/uk/gov/hmrc/gform/formtemplate/BooleanExprValidator.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.formtemplate
+
+import uk.gov.hmrc.gform.core.{ Invalid, Valid, ValidationResult }
+import uk.gov.hmrc.gform.core.ValidationResult.BooleanToValidationResultSyntax
+import uk.gov.hmrc.gform.sharedmodel.formtemplate._
+
+class BooleanExprValidator(
+  indexLookup: Map[FormComponentId, Int],
+  wrapperType: BooleanExprWrapperType
+)(
+  idx: Int
```

Comment:
probably worth renaming idx to pageIdx for clarity?

Reply:
sure